torchvision/utils.py

```diff
@@ -362,6 +364,11 @@ def draw_bounding_boxes(
     else:
         label_colors = colors.copy()  # type: ignore[assignment]
 
+    if fill_labels:
+        label_background_colors = _parse_colors(label_background_colors, num_objects=num_boxes) if label_background_colors else colors.copy()  # type: ignore[assignment]
+    else:
+        label_background_colors = colors.copy()  # type: ignore[assignment]
```

Comment:
Maybe we can add the conditions ` if label_background_colors` in the first `if fill_labels` for more clarity here?